cpp/arcticdb/column_store/column.cpp

```diff
@@ -220,66 +220,65 @@ void Column::append(const Column& other, position_t at_row) {
                 row_count(), sparse_map().count());
 }
 
+void Column::physical_sort_external(std::vector<uint32_t> &&sorted_pos) {
+    size_t physical_rows = row_count();
+    auto& buffer = data_.buffer();
+
+    util::check(sorted_pos.size() == physical_rows, "Mismatch between sorted_pos size and row_count: {} != {}",
+        sorted_pos.size(), physical_rows);
+
+    type().visit_tag([&buffer, &sorted_pos, &physical_rows] (auto tdt) {
+        using TagType = decltype(tdt);
+        using RawType = typename TagType::DataTypeTag::raw_type;
+
+        for (auto i=0u; i<physical_rows; ++i){
+            if (i != sorted_pos[i]){
+                auto& current = buffer.cast<RawType>(i);
+                // Amortized O(1) complexity, because each iteration places an element where it's supposed to go
+                // and once an element is in it's sorted position we never move it.
+                while (i != sorted_pos[i]){
+                    auto move_to = sorted_pos[i];
+                    std::swap(sorted_pos[i], sorted_pos[move_to]);
```

Comment:
Can you explain briefly how much difference this optimization makes, and why? It seems like it is introducing a fair bit of branching in the case where the numbers are randomly distributed